File: 448_Find_All_Numbers_Disappeared_in_an_Array/448_1.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for n in nums:
            i = abs(n) - 1
            if nums[i] > 0:
                nums[i] = -nums[i]
        return [i + 1 for i, n in enumerate(nums) if n > 0]

File: 448_Find_All_Numbers_Disappeared_in_an_Array/448_1.scala
object Solution {
  def findDisappearedNumbers(nums: Array[Int]): List[Int] = {
    val n = nums.length
    nums.foreach { num =>
      val index = (num - 1) % n
      nums(index) += n
    }
    (for ((num, i) <- nums.zipWithIndex; if num <= n) 
      yield i + 1).toList
  }
}
